doc/configuration.rst

```diff
@@ -184,6 +185,31 @@ As the patterns are parsed as `regular expressions`_, users are advised to consu
 
         $ sphinx-build -D sphinx_gallery_conf.filename_pattern=plot_specific_example\.py ...
 
+.. _reset_argv:
+
+Passing command line arguments to example scripts
+=================================================
+
+By default, Sphinx-Gallery will not pass any command line arguments to example scripts.
+By setting the ``reset_argv`` option, it is possible to change this behavior and pass command line arguments to example scripts.
+``reset_argv`` needs to be a Callable that accepts the ``gallery_conf`` and ``script_vars`` dictionaries and returns a list of strings that are passed as additional command line arguments to the interpreter.
+
+An example could be::
+
+    def reset_argv(gallery_confg, script_vars):
+        if script_vars['src_file'] == 'example1.py':
+	    return ['-a', '1']
+	elif script_vars['src_file'] == 'example2.py':
+	    return ['-a', '2']
+
```

Comment:
Just to be explicit could we also add an example of how to set the conf, underneath defining the function, e.g.,:

```
    sphinx_gallery_conf = {
        ...
        'reset_argv': reset_argv(gallery_confg, script_vars),
    }
```
(not sure that I've done it right there)

Reply:
It would just be:
```
    sphinx_gallery_conf = {
        ...
        'reset_argv': reset_argv,
    }
```

Reply:
> It would just be:

:sweat: that's what I thought initially

Reply:
Thanks, updated in fcf0db3ba3253cf0e0b793a2dd4f395296d9a968 and a674033f414ccb15434b4a52b6af0ad171ec15d5

Reply:
I also changed the example from a function to a class. However, @larsoner , could you explain why a stable `__repr__` is important? Is this used by sphinx-gallery internally to determine if there were any changes to the code?